Install script failing on D1 creation

I fixed a couple of issues with the install script in #174 but gave up on the third and did the rest manually. The script is failing to correctly parse the response from Cloudflare when creating the D1 database. The db is correctly created, but the script doesn't parse the id.

```
[INFO] Creating D1 database: rushomon-production
[ERROR] Failed to create database: rushomon-production - no ID returned
[ERROR] Failed to create/get D1 database
[ERROR] Deployment failed
[ERROR] Script failed with exit code 1
```

--- a/config/setup.example.yaml
+++ b/config/setup.example.yaml
@@ -29,17 +29,37 @@ oauth:
   # GitHub OAuth (recommended, widely used)
   github:
     enabled: true
-    client_id: "Iv1.abc123..."
-    # Secret can be environment variable or inline (environment variable recommended)
-    # Set GITHUB_CLIENT_SECRET environment variable before running setup
+    client_id: "Ov23lixxxxxxxxxxxxxxx"  # Your GitHub OAuth App Client ID
+    # Secret should be set via GITHUB_CLIENT_SECRET environment variable
     client_secret: "${GITHUB_CLIENT_SECRET}"
 
   # Google OAuth (optional)
   google:
     enabled: false
-    client_id: ""
-    # Set GOOGLE_CLIENT_SECRET environment variable if enabled
+    client_id: ""  # Your Google OAuth Client ID
+    # Secret should be set via GOOGLE_CLIENT_SECRET environment variable
     client_secret: "${GOOGLE_CLIENT_SECRET}"
+
+# Email configuration (optional - for team invitations)
+mailgun:
+  enabled: false
+  domain: ""  # Your Mailgun domain (e.g., mg.yourdomain.com)
+  base_url: "https://api.mailgun.net/v3"
+  from: "Rushomon <noreply@yourdomain.com>"
+  # API key should be set via MAILGUN_API_KEY environment variable
+
+# Billing configuration (optional - for paid plans)
+polar:
+  enabled: false
+  # Access token from Polar.sh dashboard settings
+  access_token: "${POLAR_ACCESS_TOKEN}"
+  # Webhook secret from Polar.sh webhook configuration
+  webhook_secret: "${POLAR_WEBHOOK_SECRET}"
+  # Product IDs from Polar.sh dashboard (leave empty if not using that plan)
+  pro_monthly_product_id: ""
+  pro_annual_product_id: ""
+  business_monthly_product_id: ""
+  business_annual_product_id: ""
 
 # Deployment configuration
 deployment:
@@ -76,6 +96,10 @@ advanced:
   # Leave empty to skip R2 backups
   r2_backup_bucket: ""
 
+  # R2 bucket for organization logo storage (required for QR code logo feature)
+  # Will be created automatically if it doesn't exist
+  r2_assets_bucket_name: "rushomon-production-assets"
+
 # Example configurations for different scenarios:
 
 # Example 1: Single domain deployment (simplest)
@@ -108,10 +132,37 @@ advanced:
 #   api: "staging.rushomon.example.com"
 #   redirect: "staging.rushomon.example.com"
 
+# Example 5: Production with all features enabled
+# oauth:
+#   github:
+#     enabled: true
+#     client_id: "Ov23lixxxxxxxxxxxxxxx"
+#     client_secret: "${GITHUB_CLIENT_SECRET}"
+#   google:
+#     enabled: true
+#     client_id: "619822820604-a52lapfu2gad1738t9nd5sog7bmsbqr2.apps.googleusercontent.com"
+#     client_secret: "${GOOGLE_CLIENT_SECRET}"
+# mailgun:
+#   enabled: true
+#   domain: "mg.rushomon.example.com"
+#   from: "Rushomon <noreply@rushomon.example.com>"
+# polar:
+#   enabled: true
+#   pro_monthly_product_id: "prod_xxxxxxxxxxxxxxxx"
+#   pro_annual_product_id: "prod_xxxxxxxxxxxxxxxx"
+#   business_monthly_product_id: "prod_xxxxxxxxxxxxxxxx"
+#   business_annual_product_id: "prod_xxxxxxxxxxxxxxxx"
+# advanced:
+#   r2_assets_bucket_name: "rushomon-assets"
+
 # Usage:
 # 1. Copy this file: cp config/setup.example.yaml config/setup.yaml
 # 2. Edit config/setup.yaml with your values
 # 3. Set environment variables for secrets:
-#    export GITHUB_CLIENT_SECRET="your-secret"
+#    export GITHUB_CLIENT_SECRET="your-github-secret"
+#    export GOOGLE_CLIENT_SECRET="your-google-secret"     # If Google OAuth enabled
 #    export JWT_SECRET="your-jwt-secret"
+#    export MAILGUN_API_KEY="your-mailgun-key"            # If Mailgun enabled
+#    export POLAR_ACCESS_TOKEN="your-polar-token"         # If Polar billing enabled
+#    export POLAR_WEBHOOK_SECRET="your-polar-webhook"     # If Polar billing enabled
 # 4. Run setup: ./scripts/setup.sh --config config/setup.yaml

--- a/scripts/lib/cloudflare.sh
+++ b/scripts/lib/cloudflare.sh
@@ -11,8 +11,8 @@
 #   - Resource management
 #
 
-# Suppress wrangler update warnings that break jq JSON parsing
-export WRANGLER_LOG="error"
+# Note: WRANGLER_LOG="error" is set per-command to suppress warnings that break JSON parsing
+# Do not set it globally as it interferes with JSON output parsing
 
 # Source common utilities if not already loaded
 if [ -z "$BLUE" ]; then
@@ -118,13 +118,50 @@ EOF
   rm -f "$temp_config"
 
   debug "Raw D1 create output: $output"
+  debug "Output length: ${#output}"
+  debug "Output first 200 chars: ${output:0:200}"
 
-  # Extract database_id from the JSON output
-  local db_id=$(echo "$output" | tr -d '\033' | grep -o '"database_id": "[^"]*"' | sed 's/"database_id": "//;s/"//' | tail -1)
+  # Try multiple extraction methods for maximum compatibility
+  local db_id=""
+
+  # Method 1: Extract database_id from wrangler output format
+  if [ -z "$db_id" ]; then
+    db_id=$(echo "$output" | grep 'database_id = ' | sed 's/.*database_id = "//;s/".*//' | head -1)
+    debug "Method 1 (wrangler format): ${db_id:-"EMPTY"}"
+  fi
+
+  # Method 2: Extract database_id from JSON format (fallback)
+  if [ -z "$db_id" ]; then
+    db_id=$(echo "$output" | tr -d '\033' | jq -r '.database_id // empty' 2>/dev/null || echo "")
+    debug "Method 2 (JSON format): ${db_id:-"EMPTY"}"
+  fi
+
+  # Method 3: Extract JSON objects then parse (fallback)
+  if [ -z "$db_id" ] || [ "$db_id" = "null" ]; then
+    db_id=$(echo "$output" | tr -d '\033' | grep -o '{[^}]*}' | jq -r '.database_id // empty' 2>/dev/null | tail -1 || echo "")
+    debug "Method 3 (extract JSON): ${db_id:-"EMPTY"}"
+  fi
+
+  # Method 4: Fallback to original grep/sed method
+  if [ -z "$db_id" ] || [ "$db_id" = "null" ]; then
+    db_id=$(echo "$output" | tr -d '\033' | grep -o '"database_id": "[^"]*"' | sed 's/"database_id": "//;s/"//' | tail -1)
+    debug "Method 4 (grep/sed fallback): ${db_id:-"EMPTY"}"
+  fi
+
+  debug "Final db_id: ${db_id:-"EMPTY"}"
 
   if [ -z "$db_id" ] || [ "$db_id" = "null" ]; then
+    # Enable debug output for troubleshooting
+    export DEBUG=true
     error "Failed to create database: $db_name - no ID returned"
     debug "Output was: $output"
+    debug "Output length: ${#output}"
+    debug "Output first 500 chars: ${output:0:500}"
+    echo ""
+    echo -e "${YELLOW}=== DEBUGGING INFO ===${NC}"
+    echo "This appears to be the issue reported in #175."
+    echo "Please share this output with the developers to help fix the parsing issue."
+    echo ""
     return 1
   fi
 
@@ -166,7 +203,7 @@ list_kv_namespaces() {
 kv_namespace_exists() {
   local kv_title="$1"
 
-  local existing_id=$(wrangler kv namespace list 2>/dev/null | tr -d '\033' | jq -r ".[] | select(.title == \"$kv_title\") | .id")
+  local existing_id=$(wrangler kv namespace list 2>/dev/null | sed -n '/^\[/,$p' | jq -r ".[] | select(.title == \"$kv_title\") | .id" 2>/dev/null || echo "")
 
   if [ -n "$existing_id" ] && [ "$existing_id" != "null" ]; then
     return 0
@@ -184,13 +221,11 @@ get_kv_namespace_id() {
     return 1
   }
 
-  local kv_id=$(echo "$output" | tr -d '\033' | jq -r ".[] | select(.title == \"$kv_title\") | .id" 2>/dev/null) || {
-    error "Failed to parse KV namespace list"
-    return 1
-  }
+  local kv_id=$(echo "$output" | sed -n '/^\[/,$p' | jq -r ".[] | select(.title == \"$kv_title\") | .id" 2>/dev/null || echo "")
 
   if [ -z "$kv_id" ] || [ "$kv_id" = "null" ]; then
     error "KV namespace '$kv_title' not found"
+    echo "$output"
     return 1
   fi
 
@@ -232,13 +267,50 @@ EOF
   rm -f "$temp_config"
 
   debug "Raw KV create output: $output"
+  debug "Output length: ${#output}"
+  debug "Output first 200 chars: ${output:0:200}"
 
-  # Extract id from the JSON output
-  local kv_id=$(echo "$output" | tr -d '\033' | grep -o '"id": "[^"]*"' | sed 's/"id": "//;s/"//' | tail -1)
+  # Try multiple extraction methods for maximum compatibility
+  local kv_id=""
+
+  # Method 1: Extract id from wrangler output format
+  if [ -z "$kv_id" ]; then
+    kv_id=$(echo "$output" | grep 'id = ' | sed 's/.*id = "//;s/".*//' | head -1)
+    debug "Method 1 (wrangler format): ${kv_id:-"EMPTY"}"
+  fi
+
+  # Method 2: Extract id from JSON format (fallback)
+  if [ -z "$kv_id" ]; then
+    kv_id=$(echo "$output" | tr -d '\033' | jq -r '.id // empty' 2>/dev/null || echo "")
+    debug "Method 2 (JSON format): ${kv_id:-"EMPTY"}"
+  fi
+
+  # Method 3: Extract JSON objects then parse (fallback)
+  if [ -z "$kv_id" ] || [ "$kv_id" = "null" ]; then
+    kv_id=$(echo "$output" | tr -d '\033' | grep -o '{[^}]*}' | jq -r '.id // empty' 2>/dev/null | tail -1 || echo "")
+    debug "Method 3 (extract JSON): ${kv_id:-"EMPTY"}"
+  fi
+
+  # Method 4: Fallback to original grep/sed method
+  if [ -z "$kv_id" ] || [ "$kv_id" = "null" ]; then
+    kv_id=$(echo "$output" | tr -d '\033' | grep -o '"id": "[^"]*"' | sed 's/"id": "//;s/"//' | tail -1)
+    debug "Method 4 (grep/sed fallback): ${kv_id:-"EMPTY"}"
+  fi
+
+  debug "Final kv_id: ${kv_id:-"EMPTY"}"
 
   if [ -z "$kv_id" ] || [ "$kv_id" = "null" ]; then
+    # Enable debug output for troubleshooting
+    export DEBUG=true
     error "Failed to create KV namespace: $kv_title - no ID returned"
     debug "Output was: $output"
+    debug "Output length: ${#output}"
+    debug "Output first 500 chars: ${output:0:500}"
+    echo ""
+    echo -e "${YELLOW}=== DEBUGGING INFO ===${NC}"
+    echo "This appears to be a similar parsing issue to #175."
+    echo "Please share this output with the developers to help fix the parsing issue."
+    echo ""
     return 1
   fi
 
@@ -380,8 +452,8 @@ list_zones() {
 create_or_get_r2_bucket() {
   local bucket_name="$1"
 
-  # Check if bucket exists
-  if wrangler r2 bucket list --json 2>/dev/null | jq -e ".[] | select(.name == \"$bucket_name\")" >/dev/null; then
+  # Check if bucket exists (parse human-readable output)
+  if wrangler r2 bucket list 2>/dev/null | grep -q "^name:.*$bucket_name$"; then
     info "R2 bucket '$bucket_name' already exists"
     return 0
   fi

--- a/scripts/lib/common.sh
+++ b/scripts/lib/common.sh
@@ -195,6 +195,7 @@ load_config_with_yq() {
   # Advanced options
   export ENABLE_KV_RATE_LIMITING=$(yq eval '.advanced.enable_kv_rate_limiting // false' "$config_file")
   export R2_BACKUP_BUCKET=$(yq eval '.advanced.r2_backup_bucket // ""' "$config_file")
+  export R2_ASSETS_BUCKET_NAME=$(yq eval '.advanced.r2_assets_bucket_name // ""' "$config_file")
 
   # Cloudflare
   export CLOUDFLARE_ACCOUNT_ID=$(yq eval '.cloudflare.account_id // ""' "$config_file")
@@ -211,7 +212,10 @@ load_config_basic() {
   export GITHUB_CLIENT_ID=$(grep 'client_id:' "$config_file" | head -1 | awk '{print $2}' | tr -d '"' || echo "")
   export ENVIRONMENT_NAME=$(grep 'environment_name:' "$config_file" | awk '{print $2}' | tr -d '"' || echo "production")
   export WORKER_NAME=$(grep 'worker_name:' "$config_file" | awk '{print $2}' | tr -d '"' || echo "rushomon-production")
-  
+
+  # R2 assets bucket (basic parsing)
+  export R2_ASSETS_BUCKET_NAME=$(grep 'r2_assets_bucket_name:' "$config_file" | awk '{print $2}' | tr -d '"' || echo "")
+
   # Cloudflare configuration
   export CLOUDFLARE_ACCOUNT_ID=$(grep 'account_id:' "$config_file" | awk '{print $2}' | tr -d '"' || echo "")
 
@@ -270,9 +274,26 @@ deployment:
 secrets:
   jwt_secret: "\${JWT_SECRET}"
 
+mailgun:
+  enabled: $([ -n "${MAILGUN_DOMAIN:-}" ] && echo "true" || echo "false")
+  domain: "${MAILGUN_DOMAIN:-}"
+  base_url: "${MAILGUN_BASE_URL:-}"
+  from: "${MAILGUN_FROM:-}"
+  # Note: api_key should be set via MAILGUN_API_KEY environment variable
+
+polar:
+  enabled: ${POLAR_ENABLED:-false}
+  access_token: "\${POLAR_ACCESS_TOKEN}"
+  webhook_secret: "\${POLAR_WEBHOOK_SECRET}"
+  pro_monthly_product_id: "${POLAR_PRO_MONTHLY_PRODUCT_ID:-}"
+  pro_annual_product_id: "${POLAR_PRO_ANNUAL_PRODUCT_ID:-}"
+  business_monthly_product_id: "${POLAR_BUSINESS_MONTHLY_PRODUCT_ID:-}"
+  business_annual_product_id: "${POLAR_BUSINESS_ANNUAL_PRODUCT_ID:-}"
+
 advanced:
   enable_kv_rate_limiting: ${ENABLE_KV_RATE_LIMITING:-false}
   r2_backup_bucket: "${R2_BACKUP_BUCKET:-}"
+  r2_assets_bucket_name: "${R2_ASSETS_BUCKET_NAME:-}"
 EOF
 
   success "Configuration saved to $config_file"

--- a/scripts/setup.sh
+++ b/scripts/setup.sh
@@ -15,8 +15,15 @@
 #
 # Examples:
 #   ./scripts/setup.sh
+#   ./scripts/setup.sh --dry-run
 #   ./scripts/setup.sh --config config/staging.yaml
 #   ./scripts/setup.sh --config config/production.yaml --update
+#   ./scripts/setup.sh --config config/staging.yaml --dry-run  # Preview existing config
+#
+# Notes:
+#   - --dry-run generates config/{environment}.yaml when no --config is provided
+#   - --dry-run shows existing config when --config is provided (no overwriting)
+#   - Use environment variables for sensitive data (secrets, API keys)
 #
 
 set -euo pipefail
@@ -56,6 +63,13 @@ export MAILGUN_DOMAIN=""
 export MAILGUN_BASE_URL=""
 export MAILGUN_FROM=""
 export MAILGUN_API_KEY=""
+export POLAR_ENABLED=false
+export POLAR_ACCESS_TOKEN=""
+export POLAR_WEBHOOK_SECRET=""
+export POLAR_PRO_MONTHLY_PRODUCT_ID=""
+export POLAR_PRO_ANNUAL_PRODUCT_ID=""
+export POLAR_BUSINESS_MONTHLY_PRODUCT_ID=""
+export POLAR_BUSINESS_ANNUAL_PRODUCT_ID=""
 export ENABLE_KV_RATE_LIMITING=false
 export R2_BACKUP_BUCKET=""
 export R2_ASSETS_BUCKET_NAME=""
@@ -503,6 +517,31 @@ main() {
   # Dry run mode
   if [ "$DRY_RUN" = true ]; then
     info "DRY RUN MODE - No changes will be made"
+
+    if [ -n "$CONFIG_FILE" ]; then
+      info "Using existing configuration file: $CONFIG_FILE"
+      info "Configuration preview:"
+      echo ""
+      cat "$CONFIG_FILE"
+      echo ""
+      info "To run the actual deployment: ./scripts/setup.sh --config $CONFIG_FILE"
+    else
+      # Generate configuration file for dry-run
+      local config_dir="$PROJECT_ROOT/config"
+      mkdir -p "$config_dir"
+
+      local config_file="$config_dir/${ENVIRONMENT_NAME}.yaml"
+      info "Generating configuration file: $config_file"
+
+      if save_config "$config_file"; then
+        success "Configuration file generated: $config_file"
+        info "You can run the actual deployment with: ./scripts/setup.sh --config $config_file"
+      else
+        error "Failed to generate configuration file"
+        exit 1
+      fi
+    fi
+
     exit 0
   fi
 

